homeassistant/components/weatherflow_cloud/sensor.py

```diff
@@ -176,35 +280,165 @@ async def async_setup_entry(
 ) -> None:
     """Set up WeatherFlow sensors based on a config entry."""
 
-    coordinator: WeatherFlowCloudDataUpdateCoordinator = hass.data[DOMAIN][
-        entry.entry_id
-    ]
+    coordinators: WeatherFlowCoordinators = hass.data[DOMAIN][entry.entry_id]
+    rest_coordinator = coordinators.rest
+    wind_coordinator: WeatherFlowCloudDataCallbackCoordinator[
+        EventDataRapidWind, EventDataRapidWind, RapidWindListenStartMessage
+    ] = coordinators.wind
+    observation_coordinator: WeatherFlowCloudDataCallbackCoordinator[
+        WebsocketObservation, WebsocketObservation, ListenStartMessage
+    ] = coordinators.observation
 
-    async_add_entities(
-        WeatherFlowCloudSensor(coordinator, sensor_description, station_id)
-        for station_id in coordinator.data
+    entities: list[SensorEntity] = [
+        WeatherFlowCloudSensorREST(rest_coordinator, sensor_description, station_id)
+        for station_id in rest_coordinator.data
         for sensor_description in WF_SENSORS
+    ]
+
+    entities.extend(
+        WeatherFlowWebsocketSensorWind(
+            coordinator=wind_coordinator,
+            description=sensor_description,
+            station_id=station_id,
+            device_id=device_id,
+        )
+        for station_id in wind_coordinator.stations.station_outdoor_device_map
+        for device_id in wind_coordinator.stations.station_outdoor_device_map[
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_WIND_SENSORS
+    )
+
+    entities.extend(
+        WeatherFlowWebsocketSensorObservation(
+            coordinator=observation_coordinator,
+            description=sensor_description,
+            station_id=station_id,
+            device_id=device_id,
+        )
+        for station_id in observation_coordinator.stations.station_outdoor_device_map
+        for device_id in observation_coordinator.stations.station_outdoor_device_map[
+            station_id
+        ]
+        for sensor_description in WEBSOCKET_OBSERVATION_SENSORS
     )
+    async_add_entities(entities)
 
 
-class WeatherFlowCloudSensor(WeatherFlowCloudEntity, SensorEntity):
-    """Implementation of a WeatherFlow sensor."""
+# Define type aliases for the observation types
+type ObservationType = EventDataRapidWind | WebsocketObservation
+type MessageType = RapidWindListenStartMessage | ListenStartMessage
+type ResponseType = RapidWindWS | WebsocketObservation
+
+# Define a type alias for the callback coordinator
+type WeatherFlowCallback = WeatherFlowCloudDataCallbackCoordinator[
+    ObservationType, MessageType, ResponseType
+]
```

Comment:
I don't understand the generics stuff happening here? Why do we use generics if the type is not narrowed down?

Reply:
I'll refactor it -> I think I got caught up in "Oh generics look cool - and python 3.13 has more features for them". Also I was hoping to reduce the amount of code